lib/encoder.js

```diff
@@ -906,6 +911,9 @@ Encoder.prototype.encodeMap = function (value, subtypes) {
     //include item
```

Comment:
From what I can tell a key cannot be undefined or null right?  Can't seem to produce a scenario where it is, so makes sense not to have to handle it.